### ISSUE TITLE
2.0.0a1 - Comments cause inconsistent indentation of CASE

### ISSUE THREAD

**Issue by alanmcruickshank** [sqlfluff/sqlfluff#4223]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Given this query:

```sql
select
    case
        when a
        then b
    end as foo,
    case
        when a -- bar
        then b
    end as bar
from c
```

the fix is:

```sql
select
    case
        when a
            then b
    end as foo,
    case
        when
            a -- bar
        then b
    end as bar
from c
```

### Expected Behaviour

The comment causes a different treatment of the fix. This is true with and without implicit indents. The indentation without the comment is the "correct" one (at least as per https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md#example-sql).

### Observed Behaviour

see above.

### How to reproduce

`sqlfluff fix -r L003 -x f -d ansi test.sql --logger rules -vvvv`

### Dialect

snowflake

### Version

2.0.0a1

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
